catboost/python-package/catboost/core.py

```diff
@@ -1476,6 +1492,13 @@ def set_feature_names(self, feature_names):
         '''
         self._object._set_feature_names(feature_names)
 
+
+    def _get_tags(self):
+        tags = _CATBOOST_SKLEARN_COMPAT_TAGS
```

Comment:
Please build the map here instead of building it outside, because it is needed only in this function. Also there is a bunch of fields that are dependant on parameter values.
Also please return every time a new map, otherwise you are changing the same map.